src/x/parsers/metric_parsers_test.go

```diff
@@ -0,0 +1,124 @@
+package parsers
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/m3db/m3/src/metrics/metric/id"
+	"github.com/m3db/m3/src/x/ident"
+	"github.com/m3db/m3/src/x/pool"
+	"github.com/m3db/m3/src/x/serialize"
+	"github.com/stretchr/testify/require"
+)
+
+func newTestID(t *testing.T, tags map[string]string) id.ID {
+	tagEncoderPool := serialize.NewTagEncoderPool(serialize.NewTagEncoderOptions(),
+		pool.NewObjectPoolOptions().SetSize(1))
+	tagEncoderPool.Init()
+
+	var stringTags []ident.Tag
+	tagNames := make([]string, 0, len(tags))
+	for name, _ := range tags {
+		tagNames = append(tagNames, name)
+	}
+	sort.Strings(tagNames)
+	for _, name := range tagNames {
+		value := tags[name]
+		stringTags = append(stringTags, ident.StringTag(name, value))
+	}
+
+	tagEncoder := tagEncoderPool.Get()
+	err := tagEncoder.Encode(ident.NewTagsIterator(ident.NewTags(stringTags...)))
+	require.NoError(t, err)
+
+	data, ok := tagEncoder.Data()
+	require.True(t, ok)
+
+	size := 1
+	tagDecoderPool := serialize.NewTagDecoderPool(
+		serialize.NewTagDecoderOptions(serialize.TagDecoderOptionsConfig{
+			CheckBytesWrapperPoolSize: &size,
+		}),
+		pool.NewObjectPoolOptions().SetSize(size))
+	tagDecoderPool.Init()
+
+	tagDecoder := tagDecoderPool.Get()
+
+	iter := serialize.NewMetricTagsIterator(tagDecoder, nil)
+	iter.Reset(data.Bytes())
+	return iter
+}
+
+func copyTags(tags map[string]string) map[string]string {
+	copy := make(map[string]string)
+	for k, v := range tags {
+		copy[k] = v
+	}
+	return copy
+}
+func TestGetMetricIDWithoutLe(t *testing.T) {
```

Comment:
what about test with le? Did we miss unit test for them?

Reply:
oh i do have a test case for that (https://github.com/databricks/m3/pull/58/files#diff-d57ef064a0d520ff1415a5617057928b5d9bece29f3fb316e210d804887d6e54R88) -- i need to rename this function

Reply:
done!